backend/app/controllers/products.js

```diff
@@ -506,29 +502,51 @@ module.exports = {
                 formattedExportRow.questionType = exportData.body[i].questionType;
                 formattedExportRow.response = exportData.body[i].value;
                 formattedExportRow.choiceText = exportData.body[i].choiceText;
-
                 if (typeof exportData.body[i].meta.publication !== 'undefined') {
                     formattedExportRow.publicationLink = exportData.body[i].meta.publication.link;
                     formattedExportRow.publicationTitle = exportData.body[i].meta.publication.title;
                     formattedExportRow.publicationAuthor = exportData.body[i].meta.publication.author;
                     formattedExportRow.publicationDate = exportData.body[i].meta.publication.date;
                 }
-
                 if (typeof exportData.body[i].comment !== 'undefined' && !_.isEmpty(exportData.body[i].comment)) {
                     const commenter = yield * common.getEntity(req, exportData.body[i].comment.userId, User, 'authId');
                     formattedExportRow.commenter = commenter.firstName + ' ' + commenter.lastName;
                     formattedExportRow.commentReason = exportData.body[i].comment.reason;
                     formattedExportRow.comment = exportData.body[i].comment.text;
                 }
-
                 formattedExportRow.date = exportData.body[i].date;
 
                 formattedExportData.push(formattedExportRow);
+
+                // Get flags for each question and create a new csv
+                const flags = yield thunkQuery(
```

Comment:
This outputs a (duplicate) discussion row for every question that shows up in the survey export, even if the discussion wasn't for that answer for that question. We can either trace the assessment id back through the discussion record's taskId and enforce that it matches the survey export entry, or just take the discussion export out of the survey export loop and do a similar filter there

Reply:
Probably the first option is easier because we can join on the taskId and look up the assessmentId pretty easily